File: client/src/main/scala/stasis/client/ops/recovery/Providers.scala
package stasis.client.ops.recovery

import stasis.client.analysis.Checksum
import stasis.client.api.clients.Clients
import stasis.client.compression.Compression
import stasis.client.encryption
import stasis.client.staging.FileStaging
import stasis.client.tracking.RecoveryTracker
import stasis.layers.telemetry.TelemetryContext

final case class Providers(
  checksum: Checksum,
  staging: FileStaging,
  compression: Compression,
  decryptor: encryption.Decoder,
  clients: Clients,
  track: RecoveryTracker,
  telemetry: TelemetryContext
)

File: client-android/lib/src/main/kotlin/stasis/client_android/lib/ops/recovery/Providers.kt
package stasis.client_android.lib.ops.recovery

import stasis.client_android.lib.encryption.Decoder as EncryptionDecoder
import stasis.client_android.lib.analysis.Checksum
import stasis.client_android.lib.api.clients.Clients
import stasis.client_android.lib.compression.Compression
import stasis.client_android.lib.staging.FileStaging
import stasis.client_android.lib.tracking.RecoveryTracker

data class Providers(
    val checksum: Checksum,
    val staging: FileStaging,
    val compression: Compression,
    val decryptor: EncryptionDecoder,
    val clients: Clients,
    val track: RecoveryTracker
)
